### DISCUSSION TITLE
Spherical Precipitate inside a matrix Mesh?

### DISCUSSION THREAD

**Discussion by naveeniisc** [idaholab/moose#16479] [Q&A Meshing] [answered]
Dear Moose Experts,
I would like to create  Spherical Precipitate inside a matrix. I have checked all meshing options available in the mesh  ( https://mooseframework.inl.gov/syntax/Mesh/index.html ).  it is possible to create this kind of meshing in the moose? if **yes** then can someone guide me how can I achieve this mesh with moose?

Thanks for the assistance!
Naveen

**Comment by GregVernon** [comment, marked as answer]
Hi @naveeniisc,

One option available to you is to use Coreform Cubit - a meshing software that natively produces the Exodus mesh format that MOOSE prefers (for file-mesh anyways).   You can get a free copy of Coreform Cubit Learn for non-commercial use, or a free-trial of Coreform Cubit for commercial use by following the links below.  Once you have Coreform Cubit, there are several options for mesh generation - tetrahedral meshing, multi-block hex-meshing, or automatic "sculpt meshing" which is particularly useful for complex material science meshes -- see images below.

About Coreform Cubit https://coreform.com/products/coreform-cubit/
Coreform Cubit Learn: https://coreform.com/products/coreform-cubit/free-meshing-software/
Coreform Cubit trial: https://coreform.com/products/trial/

I'm happy to help any questions you may have regarding Coreform Cubit meshing.

Best regards,

Greg Vernon
Director of Product Management
Coreform LLC

## Images

- Tetrahedral meshing
![image](https://user-images.githubusercontent.com/868479/102394577-319e4680-3f97-11eb-861c-878466207905.png)

- Multi-block hex-meshing
![image](https://user-images.githubusercontent.com/868479/102394397-ef750500-3f96-11eb-9716-c092e7be69a9.png)

- Sculpt automatic hex-meshing
![image](https://user-images.githubusercontent.com/868479/102395581-91e1b800-3f98-11eb-8968-8b9deac78be8.png)

**Reply by naveeniisc** [reply]
Hi Greg,
Thank you for reply, yes I want the same mesh as you shown here. let me check cCoreform Cubit Learn.
Naveen

**Comment by Ali-toghraee** [comment]
HI naveeniisc,

If they are separate phases (different order parameters if you use PF) you can define it in you initial conditions using  "SpecifiedSmoothSuperellipsoidIC" . Look at "https://mooseframework.inl.gov/source/ics/SpecifiedSmoothSuperellipsoidIC.html for more details.
![c_0000](https://user-images.githubusercontent.com/77089229/109595011-f230c200-7ad0-11eb-8dc2-d5ea677ce67f.png)

I have done that in 2D.
